docs/R-IOC/r06_df_matrices.md

```diff
@@ -0,0 +1,297 @@
+
+Similar to how to create a matrix, we can bind named vectors through the function `data.frame()` to create a data frame with a mixture of numeric and character columns.
+
+```r
+my_df <- data.frame(
+  "id" = 1:5,
+  "age" = c(21, 25, 18, 35, 27),
+  "sex" = c("female", "female", "male", "male", "male"),
+  stringsAsFactors = FALSE # by default for R > 4.0
+)
+rownames(my_df) <- paste0("sample", 1:5) # data.frame can have unique rownames
+```
+
+By default, `data.frame` requires the columns are of equal length.
+If it is not the case and when it is possible, `data.frame` will recycle the elements of the shorter vector(s).
+
+```r
+data.frame(x = 1:4, y = 1:2)
+data.frame(x = 1:4, y = 1:3) # will return an error
+```
+
+You can use `rownames()` and `colnames()` to name/rename the names of row or column.
+You can use the `as.data.frame()` function to convert a `vector`, a `list` or a `matrix` into a `data.frame`.
+
+### Access and Modification of Data Frames
+
+Similar to the way that we use for matrix, we can access the elements in a data.frame by using the index or the name of rows or columns.
+
+```r
+my_df[, c(2, 3)] # get the 2nd and the 3rd columns by column index
+my_df[c(2, 3)] # the same as above but NOT suggested
+
+my_df[, c("age", "sex")] # get the 2nd and the 3rd columns by colnames
+my_df[c("age", "sex")] # the same as above but NOT suggested
+```
+
+We can inverse select the column by adding `-` before the column index or name:
+
+```r
+my_df[, -c(2, 3)] # get all columns except the 2nd and the 3rd
+my_df[, -c("age", "sex")]
+```
+
+The same logique to access to rows:
+
+```r
+my_df[c("sample1", "sample5"), ]
+my_df[c(1, 5), ]
+
+my_df[-c("sample1", "sample5"), ]
+my_df[-c(1, 5), ]
+```
+
+Particularly for columns, the `[[` and `$` operators can be used to select a single column and return the values in a vector.
+The main difference is that `$` does not allow index, while `[[` allow both column name and index.
+
+```r
+my_df$age
+my_df[["age"]]
+my_df[[2]]
+```
+
+How to modify a `data.frame`?
+We can use `$` or `cbind` to add a named vector of equal length as the other columns or a named vector length of 1 as a new column.
+
+```r
+my_df$new_col <- letters[1:5]
+# or
+cbind(my_df, "new_col" = letters[1:5])
+my_df$new_col2 <- "cohort1"
+```
+
+To delete colummns in a `data.frame`, we can simply affect the wanted columns to `NULL`.
+
+```r
+my_df$new_col <- NULL
+```
+
+For other possible manipulations in `matrix` and `data.frame`, please refer to the sections [8.3 to 8.6](https://bookdown.org/ndphillips/YaRrr/matrix-and-dataframe-functions.html) of Philips’ book.
+
+### To Go Further
+
+Besides the traditional `data.frame` class, there are other "enhanced" version of `data.frame`, for example:
+
+* the `tibble` ([tbl-df](https://search.r-project.org/CRAN/refmans/tibble/html/tbl_df-class.html)), which is the central data structure used by differents packages from <code>[tidyverse](https://www.tidyverse.org/packages/)</code>.
```

Comment:
I might remove this part because we detailled it after ? What do you think ?

Reply:
Ok to keep it simple ;)